src/librustc/ty/context.rs

```diff
@@ -718,19 +718,19 @@ impl<'tcx> TypeckTables<'tcx> {
         }
     }
 
-    pub fn cast_kinds(&self) -> LocalTableInContext<'_, ty::cast::CastKind> {
-        LocalTableInContext {
-            local_id_root: self.local_id_root,
-            data: &self.cast_kinds
-        }
+    pub fn is_coercion_cast(&self, hir_id: &hir::HirId) -> bool {
```

Comment:
```suggestion
    pub fn is_coercion_cast(&self, hir_id: hir::HirId) -> bool {
```

Reply:
May I know why is this better than passing the reference?

Reply:
Ids are small and simple by definition. Having a reference to a `HirId` is like passing a reference to a `u64`. You could just as well pass the `u64`, which is just as cheap to pass, but saves you a dereference on the other side. We even have a clippy lint for it: https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref